ui/components/app/assets/token-list/token-list.tsx

```diff
@@ -36,18 +38,18 @@ function TokenList({ onTokenClick }: TokenListProps) {
     chainIds: chainIdsToPoll as Hex[],
   });
 
-  const nonEvmNativeToken = useNativeTokenBalance();
+  const multichainAssets = useMultiChainAssets();
 
   // network filter to determine which tokens to show in list
   // on EVM we want to filter based on network filter controls, on non-evm we only want tokens from that chain identifier
   const { networkFilter } = useNetworkFilter();
 
   const sortedFilteredTokens = useMemo(() => {
-    const balances = isEvm ? evmBalances : [nonEvmNativeToken];
+    const balances = isEvm ? evmBalances : multichainAssets;
     const filteredAssets: TokenWithFiatAmount[] = filterAssets(balances, [
       {
         key: 'chainId',
-        opts: isEvm ? networkFilter : { [nonEvmNativeToken.chainId]: true },
+        opts: isEvm ? networkFilter : { [MultichainNetworks.SOLANA]: true },
```

Comment:
Have the controller updates been integrated on extension? Once they are, we can rely on the result of `currentNetwork.chainId` rather than this hard coded string.

Reply:
updated here [21abbb2](https://github.com/MetaMask/metamask-extension/pull/30345/commits/21abbb2742994ca8255828011a4a6f77e3445dd8)